doc: document syscalls & errors

Stebalien opened the pull request:
I've decided to document these in the SDK because this is the cleanest, low-level API definition we have. My plan is to link to the documentation from the spec instead of duplicating the syscall definitions in both places.

Rendered: https://bafybeif5utc6rkqubx3iiikkpn6ubompnd7xdowliznndocyaqgozy2uni.ipfs.dweb.link/fvm_sdk/sys/index.html

anorth reviewed (approved):
Documenting here looks good. I didn't review the sets of errors described for each syscall here in detail (I assume copied faithfully from spec).

Stebalien commented:
> I assume copied faithfully from spec

Mostly. I cleaned up some descriptions (mostly removed trailing periods).

I also fixed https://github.com/filecoin-project/fvm-specs/pull/83.
